lib/ecto/type.ex

```diff
@@ -1216,6 +1216,10 @@ defmodule Ecto.Type do
     :error
   end
 
+  defp map_each([{key, nil} | t], fun, acc) do
+    map_each(t, fun, Map.put(acc, key, nil))
```

Comment:
If the type is parameterized, we should still call the type's callback, that's why we have skip_nil?, so we need similar here.

Reply:
You're absolutely correct. Just fixed it. Happy new year :)